### DISCUSSION TITLE
Trying to make a custom license stay with you.

### DISCUSSION THREAD

**Discussion by SillyDodger** [endless-sky/endless-sky#9463] [Q&A] [answered]
How do I make it so my custom license remains with the player when they switch to a new Flagship?

I'm trying to teach myself how to mod this game in a meaningful way.
Part of my idea is to build a new organization that sells their own Outfits and eventually Ships.
I wanted to gate these available purchases behind a License which you can earn after running a few missions.

I've got the missions working almost exactly the way I want. (Thanks to the super helpful documentation here on GitHub.)
I've implemented the `set "license: <my license name>"` code so it shows up in the information page under the Licenses heading.
I've created an outfit to represent the license and patterned it off of the Pilot's License outfit.
I even implemented the `outfit "my license name" 1` code so it shows up on your flagship in the Outfitter interface.

There are two things that aren't working for me.
1) My custom license isn't showing as `Installed: 1` in the Outfitter interface.
2) If I happen to sell my flagship in order to switch to a new one, the custom license disappears from the Outfitter interface.
It still shows up in the information page as a license my character has, but the Outfit is gone as far as I can tell.

I've searched for an answer for a few days now and don't see anything different about the way I've built my custom license versus how the basic Pilot's license is built.

Thanks for any suggestions or help you can provide.

Edit:  As requested, Added an example code file with the fluff removed to keep things concise.
[Endless Sky Code Example.txt](https://github.com/endless-sky/endless-sky/files/12910576/Endless.Sky.Code.Example.txt)

**Comment by lumbar527** [comment]
Name the license `"License <my license name>"`. Assign it the same way as before.

**Reply by SillyDodger** [reply]
Thank you, that got me one step closer.

Following your suggestion I placed the word `"License"` as the first part of the outfit name when defining the custom license.
I also adjusted all the points in the appropriate Missions that reference the custom license so the names were correct.

Now the custom license shows as `"installed: 1"` in the Outfitter Interface.

However when selling the old flagship in order to switch to a new one the custom license is also sold and now appears as an available outfit to purchase on the planet in which I sold the old flagship.

I noticed that the basic Pilot's License appears in every ship in your fleet, and is not something you can sell.
I may need to figure out how to recreate that.

**Reply by mOctave** [reply]
Can you post your code?

**Reply by SillyDodger** [reply]
Just updated the main post to include a sample of the code in use.

**Comment by mOctave** [comment, marked as answer]
Okay, so I've made some changes to your file. This _should_ work, although it's been a while since I've done something like this.
Further comments are in the file.
[ES Code Example Edited.txt](https://github.com/endless-sky/endless-sky/files/12911400/ES.Code.Example.Edited.txt)

**Comment by SillyDodger** [comment]
Thanks so much.

I implemented the small changes that you suggested and it worked like a charm.

In summary;  When defining an outfit with the `category "Licenses"` node you can include the word License in the name.
However when referencing the custom License within mission definitions you don't want to include the word `"License"`.

example 1: within a `to offer` node you can put `has "license: <my license name>"` or `not "license: <my license name>"`
wherein, if your custom license outfit is named `"Dodger YES License"` then you would write `has "license: Dodger YES"`

example 2: within an `action` node you can put `set "license: <my license name>"`
wherein, if your custom license outfit is named `"Dodger YES License"` then you would write `set "license: Dodger YES"`

side note: when using the `set "license: <my license name>"` you don't need to include `outfit "<my license name>" 1`
the custom license outfit will be added automatically.

The end result is that the game properly recognizes what you've created as a License which cannot be sold, and is linked more to your character than any one of your ships.  It appears on every ship you have in the Outfitter interface, and doesn't disappear when any ship is sold.

Thanks again @mOctave, that solved the issue flawlessly.
